src/Compilers/Core/Portable/Diagnostic/WellKnownDiagnosticTags.cs

```diff
@@ -48,5 +48,11 @@ public static class WellKnownDiagnosticTags
         /// specified by the 'DiagnosticId' property on 'ObsoleteAttribute'.
         /// </summary>
         public const string CustomObsolete = nameof(CustomObsolete);
+
+        /// <summary>
+        /// Indicates that the diagnostic is a compilation end diagnostic reported
+        /// from a compilation end action.
+        /// </summary>
+        public const string CompilationEnd = nameof(CompilationEnd);
```

Comment:
@dotnet/roslyn-compiler This is the only compiler change in this PR (along with the corresponding public API change below). Please see https://github.com/dotnet/roslyn/issues/47754 for more context.

Reply:
Plan is to mark this obsolete (unused) if we ever re-implement the deduplication algorithm to not need it.